geniza/corpus/management/commands/sync_transcriptions.py

```diff
@@ -0,0 +1,151 @@
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "-n",
+            "--noact",
+            action="store_true",
+            help="Do not save changes to the database",
+        )
+
+    def handle(self, *args, **options):
+        # get settings for remote git repository url and local path
+        gitrepo_url = settings.TEI_TRANSCRIPTIONS_GITREPO
+        gitrepo_path = settings.TEI_TRANSCRIPTIONS_LOCAL_PATH
+
+        # make sure we have latest tei content from git repository
+        self.sync_git(gitrepo_url, gitrepo_path)
+
+        stats = defaultdict(int)
+        # keep track of document ids with multiple digitized editions (likely merged records/joins)
+        multiedition_docs = set()
+
+        # iterate through all tei files in the repository
+        for xmlfile in glob.iglob(os.path.join(gitrepo_path, "*.xml")):
+            stats["xml"] += 1
+
+            tei = xmlmap.load_xmlobject_from_file(xmlfile, GenizaTei)
+            # some files are stubs with no content
+            # check if there is no text content; report and skip
+            if tei.no_content():
+                self.stdout.write("%s has no text content, skipping" % xmlfile)
+                stats["empty_tei"] += 1
+                continue
+
+            # get the document id from the filename (####.xml)
+            pgpid = os.path.splitext(os.path.basename(xmlfile))[0]
+            # in ONE case there is a duplicate id with b suffix on the second
+            try:
+                pgpid = int(pgpid.strip("b"))
+            except ValueError:
+                self.stderr.write("Failed to generate integer PGPID from %s" % pgpid)
+                continue
+
+            # find the document in the database
+            try:
+                doc = Document.objects.get(
+                    models.Q(id=pgpid) | models.Q(old_pgpids__contains=[pgpid])
+                )
+            except Document.DoesNotExist:
+                stats["document_not_found"] += 1
+                self.stdout.write("Document %s not found in database" % pgpid)
+                continue
+
+            if doc.fragments.count() > 1:
+                stats["joins"] += 1
+
+            footnote = None
+
+            editions = doc.footnotes.editions()
+            if editions.count() > 1:
+                # debugging output for footnote selection
+                # print('more than one edition for %s' % xmlfile)
+                # print(list(ed.source for ed in editions))
+                # try filtering by current text content
+                editions_with_content = editions.filter(content__isnull=False)
+                # print('editions with content')
+                # print(list(ed.source for ed in editions_with_content))
+                stats["multiple_editions"] += 1
+                if editions_with_content.count() > 1:
+                    stats["multiple_editions_with_content"] += 1
+                    multiedition_docs.add(doc.id)
```

Comment:
I noticed that in this case, it seems `footnote` is not set. Is that because the intended behavior for multi-edition docs isn't known yet? I guess if the edition has to be matched to a document, and there's more then one edition, it's unclear which edition should get the HTML formatted TEI content from the repo.

Reply:
correct.  I have some open questions for Rachel and Marina about how we want to determine the correct footnote (among other questions)